payjoin/src/receive/mod.rs

```diff
@@ -347,6 +348,19 @@ impl UncheckedProposal {
         self.psbt.clone().extract_tx()
     }
 
+    /// Check that the Original PSBT meet the minimum feerate receiver requires.
+    pub fn check_if_meet_min_feerate(self, receiver_min_feerate: FeeRate) -> Result<(), Error> {
+        let sender_fee_rate = self.params.min_feerate;
+        // let sender_addtional_fee_contribution = self.params.additional_fee_contribution;
```

Comment:
what else should we take into account in this check?
we have another couple of params that can be interesting:
1) `self.psbt.fee()` (this thing returns `Bitcoin::Amount` but I guess we can convert it to `bitcoin::FeeRate` 
2) `self.params.addition_fee_contr` - this need to be validated(other index input should be set) first which we do in a different check

Reply:
`params.min_feerate` != original psbt feerate. That parameter can be set arbitrarily. What we want to check is the effective FeeRate of `self.psbt` to make sure this is a reliable fallback if the sender disappears. Otherwise, receiver would be subject to probing attacks where the original psbt fee rate is too low for the receiver to actually ever get it included in a block in the fallback case. As you mention, `additional_fee_contribution` is handled later on and can be ignored in this check.

Perhaps instead of `check_if_meet_min_feerate` this functionality could be combined with `check_can_broadcast`. Since that is already a `testmempoolaccept` check to prevent probing, which is the basis for this issue.

Something like `check_broadcast_suitability(self, min_feerate: Option<bitcoin::FeeRate>, can_broadcast: impl Fn(&bitcoin::Transaction) -> Result<bool, Error>` could replace check_can_broadcast.

min_feerate could be enforced on `assume_interactive_receiver(self, min_feerate: Option<bitcoin::FeeRate>)`` using a shared helper function, too

As a bonus this uses our typestate machine so the check can't be so easily overlooked.

Reply:
sorry for the delay here, working on it

Reply:
thanks for the patience
I pushed a new version and split the commits.